### DISCUSSION TITLE
WDIO Appium: ECONNRESET issue for driver.reloadSession()

### DISCUSSION THREAD

**Discussion by DmitriiKrupinECM** [webdriverio/webdriverio#8573] [Q&A] [answered]
Hi all,

I experience the ECONNRESET issue every time when starting my tests on real devices. The log shows the following:
```
[0-0] 2022-07-19T08:37:58.863Z INFO webdriver: Initiate new session using the WebDriver protocol
[0-0] 2022-07-19T08:37:59.031Z INFO webdriver: [POST] [http://127.0.0.1:4723/wd/hub/session ](http://127.0.0.1:4723/wd/hub/session)
[0-0] 2022-07-19T08:37:59.033Z INFO webdriver: DATA {
[0-0]   capabilities: { alwaysMatch: { platformName: 'iOS' }, firstMatch: [ {} ] },
[0-0]   desiredCapabilities: { platformName: 'iOS' }
[0-0] }
[0-0] 2022-07-19T08:41:38.371Z INFO webdriver: COMMAND deleteSession()
[0-0] 2022-07-19T08:41:38.372Z INFO webdriver: [DELETE] http://127.0.0.1:4723/wd/hub/session/797613c6-2fe2-41a2-b69f-d32b449862a7
```
So basically, the test deletes the session (even for the very first test in the spec) without properly starting it and loses the connection afterwards.

The only command I use for sessions is the reload in my beforeTest hook:
```
  beforeTest: async function (test, context) {
    await driver.reloadSession()
  },
```
If I remove it - the ECONNRESET issue doesn't trigger. However, without it my app state doesn't get reset and the next test starts from the screen the previous one ended on.

Could y'all please advise on the best way to handle the sessions for mobile Appium tests (iOS)?

**Comment by erwinheitzman** [comment, marked as answer]
I'm unsure if you are testing a native app or awebsite but I assume you need to be using an Appium command like https://appium.io/docs/en/commands/device/app/reset-app/ but I could be wrong.

Let me know if that works.

**Reply by DmitriiKrupinECM** [reply]
thanks for your reply @erwinheitzman. 
The **driver.resetApp()** didn't work for me as a valid command, but **driver.reset()** actually did the job!
Kudos for your advice.

**Reply by christian-bromann** [reply]
@DmitriiKrupinECM note that this command is deprecated and removed soon. Please check with the Appium project what better command to use for this usecase.

**Reply by DmitriiKrupinECM** [reply]
@christian-bromann that's true, by the way upon additional research, I found the following commands suit my purpose as well (and even make the tests a bit faster):
`await driver.closeApp() ` after each test
`await driver.launchApp()` before each test
